frontend/device/sdl/device.lua

```diff
@@ -17,17 +53,20 @@ local Device = Generic:new{
     needsScreenRefreshAfterResume = no,
     hasColorScreen = yes,
     hasEinkScreen = no,
-    canOpenLink = yes,
+    canOpenLink = getLinkOpener,
     openLink = function(self, link)
+        local enabled, tool = getLinkOpener()
         if not link or type(link) ~= "string" then return end
-        return os.execute("xdg-open '"..link.."'") == 0
+        if not enabled or tool == nil then return end
```

Comment:
Unless the distinction between false and nil is meaningful (which it could hardly said to be here; a boolean concatenates just as badly as nil) I'd personally just write `or not tool`.

Reply:
ok, will do 

```lua
if not enabled or not tool or not link or type(link) ~= "string" then return end
```